MainEvent/main.cpp

```diff
@@ -94,6 +101,51 @@ void MoveRobot() {
   //Roboteq.sendAngles();
 }
 
+bool sortingMethod (Mat& a, Mat& b){
+    return a.Y > b.Y;
+}
+
+void matchingMethod(){
+  // DONE: Write function here to match the balls in the ballCoordL & ballCoordR and put them into vector of pair
+  //sorting here is inefficient(not like it really matters that much here), makes more sense to sort after balls out of bounds removed
+  //sort(ballCoordR.begin(), ballCoordR.end(), sortingMethod());
+  //sort(ballCoordL.begin(), ballCoordL.end(), sortingMethod());
+
+  vector<Mat> correctedR;
+  vector<Mat> correctedL;
+
+  for (auto lball: ballCoordL){
+    if (lball.X > leftBound){
+      correctedL.push_back(lball);
+    }
+  }
+  for (auto rball: ballCoordR){
+    if (rball.X < rightBound){
+      correctedR.push_back(rball);
+    }
+  }
+
+  sort(correctedR.begin(), correctedR.end(), sortingMethod());
+  sort(correctedL.begin(), correctedL.end(), sortingMethod());
+  //balls are now sorted by their Y values
+  //how much does sorting even do here?
```

Comment:
It can do a lot, but not the current way you're doing it. Sorting takes O(nlog(n)) and the loop below is O(n<sup>2</sup>) where n is the number of balls. If you remove the sorting the time complexity is the same and it's less time being spent overall. HOWEVER, the idea of sorting is useful. If you take my hints above, you can delete the sorting lines code entirely.